src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -0,0 +1,238 @@
+    }
+
+    @Timed
+    @DeleteMapping(path = [ID_PATH + EXTERNAL_DATABASE_TABLE])
+    override fun deleteExternalDatabaseTables(
+            @PathVariable(ID) organizationId: UUID,
+            @RequestBody tableNames: Set<String>) {
+        val tableIds = tableNames.map {
+            getExternalDatabaseObjectId(organizationId, it)
+        }.toSet()
+        tableIds.forEach { ensureObjectCanBeDeleted(it) }
+        val aclKeys = tableIds.map { AclKey(organizationId, it) }.toSet()
+        aclKeys.forEach { aclKey ->
+            ensureOwnerAccess(aclKey)
+            authorizations.deletePermissions(aclKey)
+            securableObjectTypes.deleteSecurableObjectType(aclKey)
+        }
+        edms.deleteOrganizationExternalDatabaseTables(organizationId, tableIds)
+    }
+
+    @Timed
+    @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + COLUMN_NAME_PATH + EXTERNAL_DATABASE_COLUMN])
+    override fun deleteExternalDatabaseColumn(
+            @PathVariable(ID) organizationId: UUID,
+            @PathVariable(TABLE_NAME) tableName: String,
+            @PathVariable(COLUMN_NAME) columnName: String
+    ) {
+        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
+        val columnId = getExternalDatabaseObjectId(tableId, columnName)
+        val aclKey = AclKey(organizationId, tableId, columnId)
+        ensureOwnerAccess(aclKey)
+        ensureObjectCanBeDeleted(columnId)
+        authorizations.deletePermissions(aclKey)
+        securableObjectTypes.deleteSecurableObjectType(aclKey)
+        edms.deleteOrganizationExternalDatabaseColumn(organizationId, columnId)
+    }
+
+    @Timed
+    @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + EXTERNAL_DATABASE_COLUMN])
+    override fun deleteExternalDatabaseColumns(
+            @PathVariable(ID) organizationId: UUID,
+            @PathVariable(TABLE_NAME) tableName: String,
+            @RequestBody columnNames: Set<String>
+    ) {
+        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
+        val columnIds = columnNames.map {
+            getExternalDatabaseObjectId(tableId, it)
+        }.toSet()
+        columnIds.forEach { ensureObjectCanBeDeleted(it) }
+        val aclKeys = columnIds.map { AclKey(organizationId, tableId, it) }.toSet()
+        aclKeys.forEach { aclKey ->
+            ensureOwnerAccess(aclKey)
+            authorizations.deletePermissions(aclKey)
+            securableObjectTypes.deleteSecurableObjectType(aclKey)
+        }
+        edms.deleteOrganizationExternalDatabaseColumns(organizationId, columnIds)
+    }
+
+    private fun ensureOwner(organizationId: UUID): AclKey {
+        val aclKey = AclKey(organizationId)
+        accessCheck(aclKey, EnumSet.of(Permission.OWNER))
+        return aclKey
+    }
+
+    private fun getExternalDatabaseObjectId(containingObjectId: UUID, name: String): UUID {
+        val fqn = FullQualifiedName(containingObjectId.toString(), name)
```

Comment:
I would use rather `HazelcastAclKeyReservationService.isReserved(fqn)` here and that way you don't have to add the `aclKeysMap` IMap to the class.

Reply:
+ you could write a function in `HazelcastAclKeyReservationService` to get ids in a batch for `deleteExternalDatabaseTables` and `deleteExternalDatabaseColumns`, so it will only tap into the hazelcast map 1 time